File: programas/tarea8.rb
#Tarea 8
#Convertidor de temperaturas
#Nombre: Erik Everardo Cavazos Hernandez
#Matrícula: 1811290
#Clase: Lenguajes de programación V4-V6 Jueves
#Docente: Ismael Pimentel
puts "Quiere convertir:"
puts "1. Fahrenheit -> Celcius"
puts "2. Celcius -> Fahrenheit"
accion = gets().to_i
if accion == 1 then
    print "Ingrese grados Fahrenheit: "
    cantidad = gets().to_f
elsif accion == 2 then
    print "Ingrese grados Celcius: "
    cantidad = gets().to_f
else
    puts "Opcion incorrecta"
end
unless cantidad == nil then
    case accion
        when 1 then puts "#{(cantidad - 32) / 1.8} grados Celcius"
        when 2 then puts "#{(cantidad*1.8) + 32} grados Fahrenheit"
    end
end

File: programas/tarea8.java
/*
Tarea 8
Convertidor de temperaturas
Estudiante: Erik Everardo Cavazos Hernández
Matrícula: 1811290
Clase: Lenguajes de Programación V4-V6 Jueves
Docente: Ismael Pimentel
*/
import javax.swing.JOptionPane;
public class tarea8{
    public static void main(String[] args){
        String [] botones = {"Grados Fahrenheit","Grados Celcius"};

        float cantidad = Float.parseFloat(JOptionPane.showInputDialog(null, "Escriba cantidad inicial"));
        float resultado;
        int opcion = JOptionPane.showOptionDialog (
            null, 
            "Convertir a: ", 
            "Convertir a...", 
            JOptionPane.YES_NO_CANCEL_OPTION, 
            JOptionPane.INFORMATION_MESSAGE, 
            null, 
            botones, 
            botones[0]);
        switch(opcion){
            case 0:
                resultado = (cantidad*1.8f) + 32;
                JOptionPane.showMessageDialog(null,resultado + " grados Fahrenheit");
                break;
            case 1:
                resultado = (cantidad-32)/1.8f;
                JOptionPane.showMessageDialog(null,resultado + " grados Celcius");
                break;
        }
    }
}